packages/react/src/components/ActionMenu/ActionMenu.tsx

```diff
@@ -0,0 +1,174 @@
+import React, {
+  forwardRef,
+  useState,
+  useRef,
+  useCallback,
+  useMemo,
+  useEffect
+} from 'react';
+import classnames from 'classnames';
+import { useId } from 'react-id-generator';
+import useSharedRef from '../../utils/useSharedRef';
+import type { onActionEvent } from '../ActionList/ActionListContext';
+import type Listbox from '../Listbox';
+import AnchoredOverlay from '../AnchoredOverlay';
+import ClickOutsideListener from '../ClickOutsideListener';
+
+const [ArrowDown, ArrowUp] = ['ArrowDown', 'ArrowUp'];
+
+type ActionMenuTriggerProps = Pick<
+  React.HTMLAttributes<HTMLButtonElement>,
+  'onClick' | 'onKeyUpCapture' | 'onKeyDown' | 'aria-expanded' | 'aria-haspopup'
+>;
+
+type ActionMenuTriggerFunction = (
+  props: ActionMenuTriggerProps,
+  open: boolean
+) => React.ReactElement;
+
+type ActionMenuProps = {
+  children: React.ReactElement;
+  trigger: React.ReactElement | ActionMenuTriggerFunction;
+} & Pick<React.ComponentProps<typeof AnchoredOverlay>, 'placement'> &
+  React.HTMLAttributes<HTMLElement>;
+
+const ActionMenu = forwardRef<HTMLElement, ActionMenuProps>(
+  (
+    {
+      className,
+      style,
+      trigger,
+      placement = 'bottom-start',
+      children: actionMenuList,
+      ...props
+    },
+    ref
+  ) => {
+    const [open, setOpen] = useState(false);
+    const [focusStrategy, setFocusStrategy] =
+      useState<React.ComponentProps<typeof Listbox>['focusStrategy']>('first');
+    const triggerRef = useRef<HTMLButtonElement>(null);
+    const actionMenuRef = useSharedRef<HTMLElement>(ref);
+    const actionMenuListRef = useSharedRef<HTMLElement>(
+      actionMenuList.props.ref
+    );
+    const [triggerId] = useId(1, 'menu-trigger');
+
+    const handleTriggerClick = useCallback(
+      (
+        event:
+          | React.MouseEvent<HTMLButtonElement>
+          | React.KeyboardEvent<HTMLButtonElement>
+      ) => {
+        // istanbul ignore else
+        if (!event.defaultPrevented) {
+          setOpen(!open);
+          setFocusStrategy('first');
+        }
+      },
+      [open]
+    );
+
+    const handleTriggerKeyDown = useCallback(
+      (event: React.KeyboardEvent<HTMLButtonElement>) => {
+        // istanbul ignore else
```

Comment:
nit - Not for this PR but it feels this rule is kinda useless if we ignore it most of the time

Reply:
I don't think this is a rule, but specifically for coverage. I don't consider checking the else condition in this case significant for coverage unless you do.